hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -441,6 +441,8 @@ private Stream<HoodieInstant> getCommitInstantsToArchive() throws IOException {
       Option<HoodieInstant> oldestInstantToRetainForClustering =
           ClusteringUtils.getOldestInstantToRetainForClustering(table.getActiveTimeline(), table.getMetaClient());
 
+      table.getIndex().updateMetadata(table);
+
```

Comment:
Why triggers the fix during archival?

Reply:
The archival process will archive replace commit from active timeline, once it does that , all the hudi writer will start referring default metadata index file that is **00000000000000.hashing_meta** , **check the function loadMetadata(HoodieTable table, String partition)**.  That's the reason it is necessary to trigger the update metadata function , so that it will bring latest metadata commit file in sync with 00000000000000.hashing_meta.

Reply:
cc @xiarixiaoyao , can you review the code?

Reply:
@rohan-uptycs, could the `updateMetadata` be invoked after archive or the caller of `getCommitInstantsToArchive`? IMO, `getCommitInstantsToArchive` only gets the commit instants to archive, no any update behavior. Therefore, the `updateMetadata` should not trigger in `getCommitInstantsToArchive`.

Reply:
@SteNicholas , make sense. Updated the code to call after **getCommitInstantsToArchive** .
Please review